codalab/lib/bundle_cli.py

```diff
@@ -1542,11 +1568,13 @@ def print_target_info(self, client, target, decorate, maxlines=10, fail_if_not_e
 
         if info_type == 'file':
             if decorate:
-                import base64
-                for line in client.head_target(target, maxlines):
-                    print >>self.stdout, formatting.verbose_contents_str(base64.b64decode(line)),
+                with closing(client.fetch_contents_blob(target[0], target[1], head=maxlines)) as contents:
+                    self.stdout.write(contents.read())
+                    self.stdout.flush()
             else:
-                client.cat_target(target, self.stdout)
+                with closing(client.fetch_contents_blob(target[0], target[1])) as contents:
+                    self.stdout.write(contents.read())
```

Comment:
We should stream the contents through using copyfileobj...otherwise if you cat a large file, it waits for everything to be downloaded before printing anything out.

On a related note, we should support 'head' and 'tail' in addition to 'cat'.  Add as another PR.